relay-sampling/src/lib.rs

```diff
@@ -441,9 +441,28 @@ impl FieldValueProvider for Event {
                 }
                 _ => Value::Null,
             },
-            field_name if field_name.starts_with("transaction.measurements.") => {
-                let measurement_name = &field_name["transaction.measurements.".len()..];
-                if let Some(value) = get_measurement(self, measurement_name) {
+            "event.contexts.trace.op" => match (self.ty.value(), store::get_transaction_op(self)) {
+                (Some(&EventType::Transaction), Some(op_name)) => Value::String(op_name.to_owned()),
+                _ => Value::Null,
+            },
+            "event.platform" => match self.platform.value() {
+                Some(platform) if store::is_valid_platform(platform) => {
+                    Value::String(platform.to_owned())
+                }
+                _ => Value::String("other".to_owned()),
+            },
+            field_name
+                if field_name.starts_with("event.measurements.")
+                    && field_name.ends_with(".value")
+
+                    // greater than: there must be at least one character between the prefix and
+                    // suffix. `event.measurements.value` is not valid. If we let that through, the
```

Comment:
```suggestion
                    // suffix. `event.measurements..value` is not valid. If we let that through, the
```

Reply:
neither are valid, but if we didn't have this extra condition, we would index `field_name[a..b]` in this exact case, where `b < a`, which is a panic